test/extensions/filters/network/common/fuzz/network_readfilter_fuzz_test.cc

```diff
@@ -21,29 +53,16 @@ DEFINE_PROTO_FUZZER(const test::extensions::filters::network::FilterFuzzTestCase
         // calls mutate on an input, and *not* during fuzz target execution.
         // Replaying a corpus through the fuzzer will not be affected by the
         // post-processor mutation.
-
-        // TODO(jianwendong): After extending to cover all the filters, we can use
-        // `Registry::FactoryRegistry<
-        // Server::Configuration::NamedNetworkFilterConfigFactory>::registeredNames()`
-        // to get all the filter names instead of calling `UberFilterFuzzer::filter_names()`.
-        static const auto filter_names = UberFilterFuzzer::filterNames();
-        static const auto factories = Registry::FactoryRegistry<
-            Server::Configuration::NamedNetworkFilterConfigFactory>::factories();
-        // Choose a valid filter name.
-        if (std::find(filter_names.begin(), filter_names.end(), input->config().name()) ==
-            std::end(filter_names)) {
-          absl::string_view filter_name = filter_names[seed % filter_names.size()];
-          if (filter_name != input->config().name()) {
-            // Clear old config, or unpacking non-suitable value may crash.
-            input->mutable_config()->clear_typed_config();
-            input->mutable_config()->set_name(std::string(filter_name));
-          }
+        static unsigned config_mutation_cnt = 0;
+        // mutate the config part of the fuzzer only so often.
+        static const unsigned config_mutation_limit = 1000;
+        static envoy::config::listener::v3::Filter config = mutateConfig(seed);
```

Comment:
I think this will break the ability to reproduce an issue.
Say a fuzzer attempts to run the following:
1. Iteration 1: Test case 1 - input:
```
config: <config1>
actions: <action1>
```

this test case succeeds and no issue is detected.

2. Iteration 2: Test case 2 (changing both the config and the action) - input:
```
config: <config2>
actions: <action2>
```

due to this PR the fuzzer will actually use config1, but if it detects an issue, rerunning with test-case2's input will not reproduce the issue.
If the function were to reproduce the same output given some input, then it should the problem.

One other thing to note, is that if the fuzzer creates an invalid config at the beginning of iterations-window (every 1000 iterations), then it will be "stuck" with the invalid config for that window.

An improvement to the above will probably be replacing an _invalid config_ with some known working config. Assume there are N known valid configs, if the post-processor validates the input's config and finds that it is invalid, it will resort to choosing one of the N known configs (using the seed).

Reply:
Well, the actual fuzz test is run with:
`
config: <config1>
actions: <action2>
`
in the second case and that is what is recorded by the fuzzer. So test-case2 in my opinion is reproducable safely, because the fuzzer will only store the input *after* the postprocessor has been run over it.

Well, the fuzzer may generate an invalid input, but the postprocessor will correct it using the validated input generator. Of course can this lead to an invalid input still, because the validated input generator is still missing some alternatives, like well-known regex and so on. 

As for the improvement, that is what the validated input generator tries to do, but on a general basis not limited to a fixed set.

Reply:
Well, you don't see it is deterministic, when it only allows changes to the config part every N-iteration? What is indeterministic in that? I agree that it will change user input in a row, i.e. when corpus1, corpus2, ... are feed to the fuzzer, then corpus2 and all N-2 following are modified. But strictly speaking is that also not fuzzing. 

So do you have any idea on how to resolve the issue we have at hand? Or shall we simply drop the PR and not implement keeping the config part for N iterations?